internal/zinc-core/src/main/scala/sbt/internal/inc/MemberRefInvalidator.scala

```diff
@@ -131,17 +131,18 @@ private[inc] class MemberRefInvalidator(log: Logger, logRecompileOnMacro: Boolea
       val dependent = memberRef.reverse(to)
       filteredDependencies(dependent)
     }
+
     private def filteredDependencies(dependent: Set[String]): Set[String] = {
       dependent.filter {
         case from if isScalaClass(from) =>
-          val affectedNames = usedNames.forward(from).filter(modifiedNames.isModified)
+          val affectedNames = usedNames.affectedNames(modifiedNames, from)
```

Comment:
I'd like to avoid creating the debugging String here unless that log level is enabled.

Maybe the most straight forward is to have an `hasAffectedName: Boolean` method with a short-circuiting, non String building version of `affectedNames`

Reply:
Pushed this suggested change